primitiveFTPd/src/org/primftpd/filesystem/FileUtil.java

```diff
@@ -0,0 +1,92 @@
+package org.primftpd.filesystem;
+
+import android.annotation.SuppressLint;
+import android.annotation.TargetApi;
+import android.content.Context;
+import android.net.Uri;
+import android.os.Build;
+import android.os.storage.StorageManager;
+import android.provider.DocumentsContract;
+
+import androidx.annotation.Nullable;
+import androidx.documentfile.provider.DocumentFile;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.lang.reflect.Array;
+import java.lang.reflect.Method;
+
+public final class FileUtil {
```

Comment:
Unbelievable that you took the time to write this :)
I'm not yet happy with the class name. IMO StorageManagerUtil would be better

Reply:
No no no, this is copy paste from the SO solution mentioned above! Zero credit to me!

Rename will be pushed soon...